core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -210,6 +210,16 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
     }
   }
 
+  def truncateCheckpoint(dataDir: File, topicAndPartition: TopicAndPartition, offset: Long) {
+    inLock(lock) {
+      val checkpoint = checkpoints(dataDir)
+      val existing = checkpoint.read()
+
+      if (existing.getOrElse(topicAndPartition, 0L) > offset)
```

Comment:
Yes. The patch will only add the entry if the topic is compacted. For topic that is not compacted, it will not have entry in checkpoint and therefore `existing.getOrElse(topicAndPartition, 0L) = 0 < offset`. Is this right?